src/system/SystemObject.h

```diff
@@ -193,6 +197,20 @@ class ObjectPool
 #endif
 };
 
+template <class T, unsigned int N>
+inline void ObjectPool<T, N>::Reset()
+{
+#if CHIP_SYSTEM_CONFIG_POOL_USE_HEAP
+    mObjects.clear();
+#else
+    memset(mArena.uMemory, 0, N * sizeof(T));
```

Comment:
```
static_assert(std::is_trivial<T>::value, "Types used in ObjectPool must be trivial");
```

?

Otherwise it's not safe to reset the pool using `memset()`..

Reply:
it looks, our existing object in the pool already violate this check, need to check further, it seems this is existing bug

Reply:
I'm not surprised, we have these problems all over the place.